app/controllers/orders_controller.rb

```diff
@@ -1,13 +1,24 @@
 class OrdersController < ApplicationController
+  before_action :authenticate_user!
+
   def add_to_order
-    if Order.last
-      order = Order.last
-    else
-      order = Order.create
+	  if session[:order_id] == nil || Order.find(session[:order_id]).finalized == true
```

Comment:
The reason we check for the Order and if it's finalized is if you want to make a new Order within the same session. So you don't keep adding stuff to your already finalized order. We made this work with binding.pry and manual testing as well as cucumber.  This is the logic that works.